### ISSUE TITLE
"unexpected error" when trying to publish snapshot from litmus dashboard

### ISSUE THREAD

**Issue by Dieterbe** [raintank/grafana#61]
i'm at https://portal.raintank.io/dashboard/raintank/rt-endpoint-web?var-endpoint=dieter_plaetinck_be&var-collector=All&var-protocol=http

i hit share->snapshot-> publish to snapshot.raintank.io
in network tab i see 4 /render calls, they all succeed
then in console i see
```
Exception { message: "", result: 2153644038, name: "", filename: "https://portal.raintank.io/app/app.02839f86.js", lineNumber: 12, columnNumber: 0, inner: null, data: null, stack: "kb/<@https://portal.raintank.io/app/app.02839f86.js:12:15874
v@https://portal.raintank.io/app/app.02839f86.js:12:14414
o/j@https://portal.raintank.io/app/app.02839f86.js:12:12985
i@https://portal.raintank.io/app/app.02839f86.js:12:29947
j/<@https://portal.raintank.io/app/app.02839f86.js:12:30119
Rb/this.$get</l.prototype.$eval@https://portal.raintank.io/app/app.02839f86.js:14:5281
Rb/this.$get</l.prototype.$digest@https://portal.raintank.io/app/app.02839f86.js:14:3795
Rb/this.$get</l.prototype.$apply@https://portal.raintank.io/app/app.02839f86.js:14:5539
f/j<@https://portal.raintank.io/app/app.02839f86.js:14:12796
g@https://portal.raintank.io/app/app.02839f86.js:11:23976
Ta/m.defer/c<@https://portal.raintank.io/app/app.02839f86.js:11:25978
" }
Exception { message: "", result: 2153644038, name: "", filename: "https://portal.raintank.io/app/app.02839f86.js", lineNumber: 12, columnNumber: 0, inner: null, data: null, stack: "kb/<@https://portal.raintank.io/app/app.02839f86.js:12:15874
v@https://portal.raintank.io/app/app.02839f86.js:12:14414
o/j@https://portal.raintank.io/app/app.02839f86.js:12:12985
i@https://portal.raintank.io/app/app.02839f86.js:12:29947
j/<@https://portal.raintank.io/app/app.02839f86.js:12:30119
Rb/this.$get</l.prototype.$eval@https://portal.raintank.io/app/app.02839f86.js:14:5281
Rb/this.$get</l.prototype.$digest@https://portal.raintank.io/app/app.02839f86.js:14:3795
Rb/this.$get</l.prototype.$apply@https://portal.raintank.io/app/app.02839f86.js:14:5539
f/j<@https://portal.raintank.io/app/app.02839f86.js:14:12796
g@https://portal.raintank.io/app/app.02839f86.js:11:23976
Ta/m.defer/c<@https://portal.raintank.io/app/app.02839f86.js:11:25978
" }
Object { message: "Unexpected error", severity: "error" } app.02839f86.js:12:24598
e/<() app.02839f86.js:12
bb/this.$get</<() app.02839f86.js:12
f/j<() app.02839f86.js:14
g() app.02839f86.js:11
Ta/m.defer/c<() app.02839f86.js:11
```

**Comment by nopzor1200** [comment]
Is this a totally separate issue to the 4 second timeout thing?

**Comment by Dieterbe** [comment]
seems like it. this case throws exceptions, the other didn't.

**Comment by nopzor1200** [comment]
@woodsaj thinks this may be related to multiselect functionality support (on snapshot.raintank.io vs portal.raintank.io)

**Comment by bulletfactory** [comment]
This is still not working -- Litmus is running latest multiselect functionality -- perhaps snapshot.raintank.io isn't?

**Comment by nopzor1200** [comment]
This is still problematic on app.raintank

Local snapshot works fine. Publish to snapshot.raintank.io does not work (doesn't even time out after >30seconds)

**Comment by torkelo** [comment]
this it is a problem with SSL (https), we do not have a HTTPS cert for snapshots-origin.raintank.io  , we have for the fastly hosted CDN domain name snapshot.raintank.io but not for the api endpoint grafana pushes snapshots to

**Comment by mattttt** [comment]
Adam at Packet is having issues creating a local snapshot on app.raintank as well. 

![screenshot 2015-06-25 12 10 17](https://cloud.githubusercontent.com/assets/2886187/8359260/2bdbe4c0-1b33-11e5-8afa-dd3d6cd63117.png)

**Comment by nopzor1200** [comment]
@torkelo could you take a look at this? I'm aware of the timeout issue but I think there's something else at play here

**Comment by nopzor1200** [comment]
So @torkelo says this is simply an issue of a bad certificate. Grafana issue still open. We just need to get the right certificate

**Comment by nopzor1200** [comment]
@ctdk can you help with this? basically we need a cert installed for snapshot-origin
It would probably be a good idea to take the time to chef-ize this server too.

**Comment by woodsaj** [comment]
This line also needs to be fixed.

https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/shareSnapshotCtrl.js#L32

If you try to publish a snapshot from a https site (eg. https://app.raintank.io) then the request will be blocked by the browser.
```
Mixed Content: The page at 'https://dev-portal.raintank.io/dashboard/file/rt-endpoint-ping.json?var-collector=All&var-endpoint=monkey_id_au' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://snapshots-origin.raintank.io/api/snapshots'. This request has been blocked; the content must be served over HTTPS.
```

**Comment by torkelo** [comment]
fixed in recent commit, https://github.com/grafana/grafana/commit/3e3980dbacdf02a05b42b1f9931dd7ecbe2bf207

**Comment by nopzor1200** [comment]
This is now resolved and working in app.raintank.io

**Comment by Dieterbe** [comment]
confirmed!
